qiskit/pulse/schedule.py

```diff
@@ -324,34 +331,51 @@ def _construct_filter(self, *filter_funcs: List[Callable],
             time_ranges: For example, [(0, 5), (6, 10)]
             intervals: For example, [Interval(0, 5), Interval(6, 10)]
         """
-        def only_channels(channels: Set[Channel]) -> Callable:
+        def only_channels(channels: Channel) -> Callable:
             def channel_filter(time_inst: Tuple[int, 'Instruction']) -> bool:
                 return any([chan in channels for chan in time_inst[1].channels])
             return channel_filter
 
-        def only_instruction_types(types: Iterable[abc.ABCMeta]) -> Callable:
+        def only_instruction_types(types: Optional[Iterable[abc.ABCMeta]] = None) -> Callable:
             def instruction_filter(time_inst: Tuple[int, 'Instruction']) -> bool:
-                return isinstance(time_inst[1], tuple(types))
+                try:
+                    iter(types)
+                    return isinstance(time_inst[1], tuple(types))
+                except TypeError:
+                    return isinstance(time_inst[1], types)
             return instruction_filter
 
-        def only_intervals(ranges: Iterable[Interval]) -> Callable:
+        def only_intervals(ranges: Interval) -> Callable:
             def interval_filter(time_inst: Tuple[int, 'Instruction']) -> bool:
-                for i in ranges:
-                    if all([(i.start <= ts.interval.shift(time_inst[0]).start
-                             and ts.interval.shift(time_inst[0]).stop <= i.stop)
+                try:
```

Comment:
It seems we have the iterable check code here and down below prior to calling this on line 372. Is there a reason for this?